design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,714 @@
+  ...
+```
+
+We will map the “interface” field to the network-name as follows:
+- bridge-primary-mac ⇒ net1
+- sriovnet-vlan100-secondary-mac ⇒ net2
+- sriovnet-vlan100-third-mac ⇒ net3
+
+With this mapping we can easily retrieve the PCI address, as the one with the corresponding “interface” name:
+```yaml
+k8s.v1.cni.cncf.io/network-status: |-
+    [{
+          "name": "kindnet",
+          "interface": "eth0",
+          "ips": [
+              "10.244.1.9"
+          ],
+          "mac": "3a:7e:42:fa:37:c6",
+          "default": true,
+          "dns": {}
+      },{
+          "name": "default/bridge-network",
+          "interface": "net1",
+          "mac": "aa:bb:cc:dd:ee:00",
+          "dns": {}
+      },{
+          "name": "default/sriov-network-vlan100",
+          "interface": "net2",
+          "mac": "aa:bb:cc:dd:ee:01",
+          "dns": {},
+          "device-info": {
+              "type": "pci",
+              "version": "1.0.0",
+              "pci": {
+                  "pci-address": "0000:65:00.2"
+              }
+      },{
+          "name": "default/sriov-network-vlan100",
+          "interface": "net3",
+          "mac": "aa:bb:cc:dd:ee:02",
+          "dns": {},
+          "device-info": {
+              "type": "pci",
+              "version": "1.0.0",
+              "pci": {
+                  "pci-address": "0000:65:00.3"
+              }
+          }
+      }]
+```
+
+Since we’re only interested in the SRIOV networks, the network-name-to-pci-address mapping will be:
+- sriovnet-vlan100-secondary-mac ⇒ 0000:65:00.2
+- sriovnet-vlan100-third-mac ⇒ 0000:65:00.3
+
+> **_Note_**:
+> We are relying on the fact that the “interface” naming convention in [[2]](https://github.com/kubevirt/kubevirt/blob/c4b6ae63c5a7f642ab86b0755dabca3b814ecb39/pkg/virt-controller/services/template.go#L1692) will remain the same.
+> To ensure that, we plan move this logic to a new package that will be used for all scenarios related to this annotations. Moreover, this will be covered in e2e test in case the naming logic is changed without using the new package.
+> One could suggest relying on the  MAC address as an identifier instead of relying on the indexation logic, but that is simply not good enough since the MAC address field is not mandatory field.
+
+##### Backward compatibility:
+The mapping mentioned above could fail for several reasons, among them:
+- networks-status annotation is not present on the vmi
+- networks-status annotation does not hold the PCI address information for all the SRIOV interfaces
+- Annotation is not valid
+
+Such a case can happen if the networks-status annotation is breaking the API or Multus deployed on the cluster is older than v3.7.
+In any case, in the event of a mapping failure, we will simply use the old mapping, in order to not break old VMIs on old clusters.
+
+# Appendix 1: Alternative Annotations Exposure Methods
+## 1. virt controller exposes network to pci mapping through the VMI annotations
```

Comment:
I don't see here the established communication flow considered:

 * virt-controller writes the node mapping to  the VMI status
 * virt-handler extracts the info for the current node the VMI s running on and sends it with its [VirtualMachineOptions](https://github.com/kubevirt/kubevirt/blob/c4b6ae63c5a7f642ab86b0755dabca3b814ecb39/pkg/handler-launcher-com/cmd/v1/cmd.pb.go#L334) construct over to the domain xml converter to virt-launcher
 * works the same for a migration target virt-handler
 * we also already have a status cleanup happen after migrations, which can remove the old host entry and only keep the current one. Also virt-controller could do the cleanup, since it already provides the info in the first place.
 
 Where I would agree that would not make sense:
 
  * trying to modify the VMI status in a way in virt-handler so that virt-launcher would only see relevant info. We don't need such tricks. We have VirtualMachineOptions already.

Reply:
Oh one addition: @davidvossel just reminded me that virt-launcher knows the node name already anyway, so using the Options may not even be necessary, which would make it even simpler. Just cleanup after migration would be a topic.

Reply:
I added the solution you describe at option 4, could you please check and let me know if that is what you meant?

Reply:
Yes looks pretty much accurate. I agree that there is likely a little bit more to code, but afterwards we have to care about much less regarding to any info flow.